atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraTopologyValidator.java

```diff
@@ -219,14 +219,14 @@ private Set<CassandraServer> getNewHostsWithInconsistentTopologiesFromTopologyRe
         switch (topologyResult.type()) {
             case CONSENSUS:
                 Preconditions.checkState(
-                        topologyResult.agreedTopology().isPresent(),
-                        "Expected to have a consistent topology for a CONSENSUS result, but did not.");
-                ConsistentClusterTopology topology =
-                        topologyResult.agreedTopology().get();
-                pastConsistentTopology.set(topology);
+                        topologyResult.agreedTopologies().isPresent(),
+                        "Expected to have one or more consistent topologies for a CONSENSUS result, but did not.");
+                ConsistentClusterTopologies topologies =
+                        topologyResult.agreedTopologies().get();
+                pastConsistentTopologies.set(topologies);
                 return EntryStream.of(newServersWithoutSoftFailures)
                         .removeValues(result -> result.type() == HostIdResult.Type.SUCCESS
-                                && result.hostIds().equals(topology.hostIds()))
+                                && topologies.sharesAtLeastOneHostId(result.hostIds()))
```

Comment:
Ah, you're right that this could lead to faster convergence - though I didn't think too much about this as indeed it would be resolved on the next refresh.